internal/providers/terraform/parser.go

```diff
@@ -799,9 +799,33 @@ func (p *Parser) parseTags(data map[string]*schema.ResourceData, confLoader *Con
 		}
 
 		resourceData.Tags = tags
+		resourceData.TagPropagation = p.getTagPropagationInfo(resourceData)
 	}
 }
 
+func (p *Parser) getTagPropagationInfo(resource *schema.ResourceData) *schema.TagPropagation {
+	for _, expected := range aws.ExpectedPropagations {
+		if expected.ResourceType != resource.Type {
+			continue
+		}
```

Comment:
can we make the `ExpectedPropagations` a map so we can just lookup the value rather than having to loop